server/src/main/scala/de/frosner/broccoli/signal/SignalManager.scala

```diff
@@ -0,0 +1,22 @@
+package de.frosner.broccoli.signal
+
+import sun.misc.{Signal, SignalHandler}
+
+/**
+  * Provide a way to register and unregister OS signals
+  */
+trait SignalManager {
+
+  /**
+    * Registers signal handler for the given signal
+    * @param signal
+    * @param handler
+    */
```

Comment:
The implementation doesn't allow to register the same signal twice, so please document this behaviour.  Ideally, add `@throws` fields below `@params` that document both exceptions thrown, namely `IllegalArgumentException` if the signal is already registered, and `UnsupportedOperationException` if the OS isn't supported.